src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -639,6 +639,32 @@ public void AssemblyResolve_IsNotCalledForCoreLibResources()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/43909", TestRuntimes.Mono)]
+        public void AssemblyResolve_ExceptionPropagatesCorrectly()
+        {
+            RemoteExecutor.Invoke(() => {
+                bool handlerExceptionThrown = false;
+
+                AppDomain.CurrentDomain.AssemblyResolve += (sender, e) =>
+                {
+                    throw new Exception("Failure");
```

Comment:
might be good to throw some purpose-built subclass of Exception, and then we can check for `Assert.IsTrue (e.InnerException is MySpecialException)` in the `catch` clause

Reply:
You want to use `Assert.IsAssignableFrom` but yeah, good idea.